docs/advanced/auth.md

```diff
@@ -0,0 +1,101 @@
+# Authentication to a remote node
```

Comment:
Not sure about the page title, since the remote driver works without authentication - we probably don't want to promote using unauthenticated TCP connections, but it's allowed, and for a setup that runs buildkitd and buildctl on the same CI machine, it seems like a reasonable deployment?

Unrelated, but we should also mention the `docker-container://` scheme here, happy to do that as a follow-up later.

Reply:
> but it's allowed, and for a setup that runs buildkitd and buildctl on the same CI machine, it seems like a reasonable deployment?

Yes but don't think users want to set up a local buildkit instance on an ephemeral GitHub Runner? I think they prefer to use this action to create a BuildKit container.

> Unrelated, but we should also mention the `docker-container://` scheme here, happy to do that as a follow-up later.

Yes sounds good. I guess it would use a remote context to connect to a remote docker engine instance?